lib/beaker-puppet/install_utils/puppet5.rb

```diff
@@ -0,0 +1,109 @@
+module BeakerPuppet
+  module Install
+    module Puppet5
+
+      # grab build json from the builds server
+      #
+      # @param [String] project_name Name of the project
+      # @param [String] project_sha SHA ref of the build for the project
+      #
+      # @return [String, String] URL to the build for the SHA of the project,
+      #   build json parsed into a ruby hash
+      def fetch_build_details(project_name, project_sha)
+        buildserver_url = 'http://builds.delivery.puppetlabs.net'
+        buildserver_projectsha_url = "#{buildserver_url}/#{project_name}/#{project_sha}"
+        file_folder = "#{buildserver_projectsha_url}/artifacts"
+        filename = "#{project_sha}.yaml"
+        dst_folder = Dir.pwd
+
+        # fetch & parse file
+        file_local_path = fetch_http_file(file_folder, filename, dst_folder)
+        file_hash = YAML.load_file(file_local_path)
+
+        # getting the relevant data from the file
+        logger.perf_output("YAML HASH BELOW:")
+        logger.perf_output(file_hash)
+        # pp file_hash
+        logger.perf_output("PLATFORM_DATA BELOW:")
+        logger.perf_output(file_hash[:platform_data])
+        return buildserver_projectsha_url, file_hash
+      end
+
+      # get the build artifact URL for a host
+      #
+      # @param [Host] host Host to get artifact URL for
+      # @param [Hash] build_details Details of the build in a hash
+      # @param [String] build_url URL to the build
+      #
+      # @return [String] URL to the build artifact
+      def hosts_artifact_url(host, build_details, build_url)
+        platform_key = host[:platform]
+        variant, version, arch, _ = platform_key.to_array
+        # step: map platform strings into platform_key's for the hash
+        if variant == 'debian' && arch == 'x86_64'
+          platform_key.sub!('x86_64', 'amd64')
+        elsif variant == 'fedora'
+          platform_key.sub!(version, "f#{version}")
+        end
```

Comment:
@mwaggett I'm not sure how @joshcooper feels about it, but as I'm seeing it, the question is really about the tradeoffs between either:

- being consistent across the OSes
- being consistent internally to Puppet's tooling

If we're consistent across the OS vendors, then we have to make sure that OS-specific changes are propagated across all of our tooling (which means across teams). Then, when those details change (& we can't control when that might happen), we have to update all of our projects to fit.

If we're consistent across internal Puppet tooling, then packaging becomes the place where the mapping between OS-specific details & puppet tooling takes place. Packaging would be the [adapter](https://en.wikipedia.org/wiki/Adapter_pattern) between Puppet (the company) & OS vendors. Then if something changes, the adapter can be updated by our OS-specific experts internally, & no one else has to miss a beat.

Note that we're off on a tangent of future work here. This doesn't need to block this PR merging, as pulling this case statement out would be trivial to do in the future, the only concern being a little scheduling between beaker-puppet & packaging work changes.